src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3940,6 +3952,8 @@ void G1CollectedHeap::evacuate_optional_collection_set(G1ParScanThreadStateSet*
       evacuate_next_optional_regions(per_thread_states);
       phase_times()->record_or_add_optional_evac_time((Ticks::now() - start).seconds() * 1000.0);
     }
+
+    rem_set()->complete_evac_phase(true /* has_more_than_one_evacuation_phase */);
```

Comment:
I think it would be cleaner with a local variable:
bool has_more_than_one_evacuation_phase = true;
rem_set()->complete_evac_phase(has_more_than_one_evacuation_phase);

Instead of the comment, same with remember_already_scanned_cards comment above

Reply:
I kept it as is as this is the common hotspot style for "documenting" flag parameters, and this suggestion does not seem better to me.